frontend/src/App.js

```diff
@@ -64,49 +70,69 @@ function App() {
+          setError={setError}
+          setErrorMessage={setErrorMessage}
+          setSuccess={setSuccess}
+          setSuccessMessage={setSuccessMessage} />
         </Header>
         <main style={{ minHeight: "85vh" }}>
-        <Switch>
-            <Route path="/dashboard">
-                <Dashboard setTickets={setTickets} tickets={tickets} />
-              </Route>
-              <Route path="/form">
-                <Form
-                  saving={saving}
-                  tickets={tickets}
-                  setSaving={setSaving}
-                  setError={setError}
-                  setErrorMessage={setErrorMessage}
-                  setSuccess={setSuccess}
-                  setTickets={setTickets}
-                ></Form>
-              </Route>
-              <Route exact path="/">
-                <Landing />
-              </Route>
-              <Route path="*">
-                <NotFound />
-              </Route>
-            </Switch>
-            {saving && <p>saving...</p>}
-            {/* {error === true || success === true ? ( */}
-              <Toast dismissOnClick={error && handleDismiss}
-                classNameP={success ? "success" : error ? "error" : "hidden"}
-              >
-                {success ? "Successfully saved" : errorMessage}
-              </Toast>
-            {/* ) : null} */}
+          <Switch>
+            <PrivateRoute path="/dashboard">
+              <Dashboard setTickets={setTickets} tickets={tickets} />
+            </PrivateRoute>
+            <Route path="/register">
+              <Register 
+                setSuccessMessage={setSuccessMessage}
+                setSaving={setSaving}
+                setError={setError}
+                setErrorMessage={setErrorMessage}
+                setSuccess={setSuccess}/>
+            </Route>
+            <Route path="/login" render={(props) => <Login 
+                setSaving={setSaving}
+                setSuccessMessage={setSuccessMessage}
+                setError={setError}
+                setErrorMessage={setErrorMessage}
+                setSuccess={setSuccess} {...props}/>}/>
+            <Route path="/form">
+              <Form
+                saving={saving}
+                tickets={tickets}
+                setSaving={setSaving}
+                setError={setError}
+                setErrorMessage={setErrorMessage}
+                setSuccess={setSuccess}
+                setSuccessMessage={setSuccessMessage}
+                setTickets={setTickets}
+              ></Form>
+            </Route>
+            <Route exact path="/">
+              <Landing />
+            </Route>
+            <Route path="*">
+              <NotFound />
+            </Route>
+          </Switch>
+          {saving && <p>saving...</p>}
+          {/* {error === true || success === true ? ( */}
+            <Toast dismissOnClick={error && handleDismiss}
+              classNameP={success ? "success" : error ? "error" : "hidden"}
```

Comment:
I'm not a big fan of nested ternary expression. If you do want it as a one-liner, you could write it like this

 ```javascript
{success && 'success || error && 'error' || 'hidden'}
```

Reply:
eslint yells at me if I do this? Should I do it anyways?